gnat2goto/driver/tree_walk.adb

```diff
@@ -1764,6 +1768,79 @@ package body Tree_Walk is
       Do_Type_Declaration (New_Type, Typedef);
    end Do_Itype_Reference;
 
+   -----------------------
+   -- Do_Case_Statement --
+   -----------------------
+
+   function Do_Case_Statement (N : Node_Id) return Irep is
+      Ret : constant Irep := New_Irep (I_Code_Block);
+      Value : constant Irep := Do_Expression (Expression (N));
+
+      --  Auxiliary function to create a single test case
+      --  to emplace in a condition from a list of alternative
+      --  values.
+      function Make_Case_Test (Alts : List_Id) return Irep;
+      function Make_Case_Test (Alts : List_Id) return Irep is
+         function Make_Single_Test (Alt : Node_Id) return Irep;
+         function Make_Single_Test (Alt : Node_Id) return Irep is
+            Ret : constant Irep := New_Irep (I_Op_Eq);
+            Rhs : constant Irep := Do_Expression (Alt);
+         begin
+            Set_Lhs (Ret, Value);
+            Set_Rhs (Ret, Rhs);
+            Set_Type (Ret, New_Irep (I_Bool_Type));
+            return Ret;
+         end Make_Single_Test;
+         First_Alt_Test : constant Irep := Make_Single_Test (First (Alts));
+         This_Alt : Node_Id := First (Alts);
+      begin
+         Next (This_Alt);
+         if not Present (This_Alt) then
+            return First_Alt_Test;
+         end if;
+         declare
+            Big_Or : constant Irep := New_Irep (I_Op_Or);
+         begin
+            Append_Op (Big_Or, First_Alt_Test);
+            Set_Type (Big_Or, New_Irep (I_Bool_Type));
```

Comment:
`Make_Bool_Type`, although even better we should probably just have a constant `Bool_Type := Make_Bool_Type` somewhere in this module so we don't keep recreating new ireps for all of these unnecessarily